x-pack/test/functional_embedded/plugins/iframe_embedded/server/plugin.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import Url from 'url';
+import { Plugin, CoreSetup, PluginInitializerContext } from 'src/core/server';
+
+function renderBody(iframeUrl: string) {
+  return `
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>Kibana embedded in iframe</title>
+  </head>
+  <body>
+    <iframe data-test-subj="iframe_embedded" width="1000" height="1200" src="${iframeUrl}" frameborder="0"/>
+  </body>
+</html>
+`;
+}
+export class IframeEmbeddedPlugin implements Plugin {
+  constructor(initializerContext: PluginInitializerContext) {}
+
+  public setup(core: CoreSetup) {
+    core.http.resources.register(
+      {
+        path: '/iframe_embedded',
+        validate: false,
+      },
+      async (context, request, response) => {
+        const { protocol, port, host } = core.http.getServerInfo();
+
+        const kibanaUrl = Url.format({ protocol, hostname: host, port });
```

Comment:
our `host` is [hostname](https://developer.mozilla.org/en-US/docs/Web/API/URL/hostname) since it cannot contain port https://github.com/elastic/kibana/blob/89bc3d82a176f609679d9d5d65fe772120f8b8b4/packages/kbn-config-schema/src/types/string_type.test.ts#L104

Reply:
created https://github.com/elastic/kibana/issues/69373